### PR TITLE
chore(next-example): add webauthn support w/ unstorage adapter

### PR DESCRIPTION
**Pull request by ndom91**
## ☕️ Reasoning

- Add `Passkey` provider and `@auth/unstorage-adapter` to our `apps/example/nextjs` app.
- Uses in-memory unstorage adapter, so data is deleted every deploy
	- That means that logins won't survive _that_ long, but for a demo where folks login once (register) and then maybe logout / log back in a few times to test, I think this is totally fine.

## 🧢 Checklist

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

**Comment by ndom91** [comment]
Lol 

image

File: packages/core/src/providers/webauthn.ts [hunk 1]
```diff
@@ -155,6 +155,8 @@ export interface WebAuthnConfig extends CommonProviderOptions {
     generateAuthenticationOptions: typeof generateAuthenticationOptions
     generateRegistrationOptions: typeof generateRegistrationOptions
   }
+  /** Custom label for the WebAuthn input field */
+  fieldLabel?: string
```
Comment (balazsorban44) [inline packages/core/src/providers/webauthn.ts:159]: I think this is not necessary. https://github.com/nextauthjs/next-auth/blob/e13ffb0cb8355071f84f0292639f2bb7b92976dd/packages/core/src/providers/webauthn.ts#L116

Which similar to Credentials: https://github.com/nextauthjs/next-auth/blob/234a150e2cac3bc62a9162fa00ed7cb16a105244/packages/core/src/providers/credentials.ts#L10-L13

**Review by balazsorban44 — DISMISSED** [review]

File: apps/examples/nextjs/auth.ts [hunk 2]
```diff
@@ -62,45 +52,76 @@ import GitHub from "next-auth/providers/github"
...
+    Google,
+    LinkedIn,
+    Passage,
+    Passkey({
+      fieldLabel: "Username",
+    }),
+    Pinterest,
+    Resend({
+      from: "no-reply@authjs.dev",
+    }),
```
[the remark this answers is not in the document]
Comment (ndom91) [reply apps/examples/nextjs/auth.ts]: Right, removed it now anyway based on our chat

File: packages/core/src/providers/webauthn.ts (hunk 1 above)
Comment (ndom91) [reply packages/core/src/providers/webauthn.ts:159]: Ah good eye! This is perfect

File: apps/examples/nextjs/auth.ts [hunk 3]
```diff
@@ -62,45 +52,76 @@ import GitHub from "next-auth/providers/github"
...
 
 import type { NextAuthConfig } from "next-auth"
 
+const prisma = new PrismaClient()
+
 export const config = {
   theme: {
     logo: "https://next-auth.js.org/img/logo/logo-sm.png",
   },
+  adapter: PrismaAdapter(prisma),
```
[the remark this answers is not in the document]
Comment (ndom91) [reply apps/examples/nextjs/auth.ts]: Note: working on this in the latest commits

**Comment by ndom91** [comment]
Update: still trying to use the advanced initialization to apply the `adapter` when trying to login with `webauthn`, but not with any of the oauth providers. Therefore we would avoid saving any session / user / account data into our database from oauth providers. 

See last 2 commits ([1](https://github.com/nextauthjs/next-auth/pull/9963/commits/f2be93d06f137905acd3ec475ff4325413446fc2), [2](https://github.com/nextauthjs/next-auth/pull/9963/commits/9b58702dafbafa482042f9003a2b1099993dbf52)) for latest updates in regard to this

**Review by ThangHuuVu — APPROVED** [review]

**Pull request merged**